### ISSUE TITLE
[Feature] User's patch management

### ISSUE THREAD

**Issue by strayge** [WowUp/WowUp#513]
### TL;DR

Allow for user to have patches for used addons.

### Why?

Sometimes user disagree with minor decisions of addon's maintainer.

Examples:
* addon has annoying popup / tooltip without way to disable it.
* addon forced you to use bad localization.

But at the same time user still want update that addons.

### Solution

Allow user to have their own patches.

1. Introduce simple markup syntax for patches or store patches at some additional file at addon folder
2. Detect properly marked patches / file with patches at addon folder
3. Apply patches after addon updates (mostly automatic but sometimes with user's interaction. in case of patched lines changes).  
4. ~~Show all installed patches with ability to enable / disable each.~~

#### ~~Sharing~~
* ~~simplified version~~  
~~copy/paste like Wago import~~  
* ~~full implementation~~  
~~ability to upload patches to wowup servers~~  
~~find popular user's patches and apply them in one click~~  

#### Updates: 

* With Curse alternative was manual update and then running python script to apply all patches.  
* With WowUp at the moment is no alternatives as modified addon disappears.  

#### No updates

Just manually patch addon and skip any updates until it completely broken.

**Comment by jliddev** [comment]
I think this ties in with GIt folder support if someone can find that ticket.

**Comment by strayge** [comment]
It was #71, but I'm not sure they are very linked.

At this case:
1. user often does not have git repo for addon (he just installed it from CF / WowInterface / etc).
2. there is updates from upstream which needed to me merged with local changes

Instead in #71:
1. user always has repo
2. upstream changes should not applies at all

**Comment by jliddev** [comment]
This needs to be broken out into separate features and discussed individually.

I see "detect patches" as in know that a user has edited something away from the originally installed code.

Then I see "share patches" as sort of a fork of the original addon?

Am i following?

**Comment by strayge** [comment]
the main part is to detect and preserve modifications through updates.

share is just minor thoughts about enhancement. definitely out of scope current issue.

**Comment by b-morgan** [comment]
I like this idea. I keep a .patch file (Unified diff) in the folder with an addon if I make any local changes. Having this applied automagically after the addon was updated would be very useful with a warning or error message if the patch fails or detects conflicts.

**Comment by jliddev** [comment]
Seems like it might be quite painful to try and manage a system like this. having to then keep which patches are good or outdated in line with all the addon providers. Seems like more trouble than gain.
